src/mesh.c

```
@@ -389,583 +368,295 @@ int mesh_update_routing(uint16_t source, uint16_t from_node, int16_t rssi, uint8
 
-    /* Determine if the packet is for us */
-    int result = packet_handle(&packet, 0); /* Direct reception */
+/* ============================================================================
+ * Internal helpers
+ * ============================================================================ */
 
-    if (result == 0 || result == 2)
-    {
-        /* Packet is for us or broadcast - process it */
-        mesh_process_packet(&packet, rssi, snr);
+static int mesh_add_route(uint16_t dest_id, uint16_t next_hop, uint8_t hops,
```

Clang (14.0.6)

[warning] 5 adjacent parameters of 'mesh_add_route' of convertible types are easily swapped by mistake

(bugprone-easily-swappable-parameters)